packages/client/src/runtime/core/engines/library/LibraryEngine.ts

```diff
@@ -535,6 +543,14 @@ You may have to run ${green('prisma generate')} for your changes to take effect.
       )
     }
 
+    if (error.user_facing_error.error_code === DRIVER_ADAPTER_EXTERNAL_ERROR && this.config.adapter) {
+      const id = error.user_facing_error.meta?.id
+      assertAlways(typeof id === 'number', 'Malformed external JS error received from the engine')
+      const errorRecord = this.config.adapter.errorRegistry.consumeError(id)
+      assertAlways(errorRecord, `External error with reported id was not registered`)
+      return errorRecord.error
+    }
```

Comment:
Should we test this somehow?

Reply:
https://github.com/prisma/team-orm/issues/363